bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -3451,6 +3449,28 @@ public void setStyle (TextStyle style, int start, int end) {
 	int length = text.length();
 	if (length == 0) return;
 	if (start > end) return;
+	int countNextLine = 0, noStyleCountNextLine = 0, loop = 0;
+	if (previousEnd < 0)
+		loop = previousEnd+1;
+	else
+		loop = previousEnd;
+	for (int i = loop; i < end; i++) {
+		if (originalText.charAt(i) == '\r' && originalText.charAt(i+1) =='\n')
```

Comment:
Is i + 1 necessarily < end?

Reply:
let me check it.

Reply:
you are correct, when string ends with \r, it is resulting OOB exception. So corrected it.